### PR TITLE
fix: add graphql valid object annotation to all input and output types

### PR DESCRIPTION
**Pull request by danadajian**
### :pencil: Description
- Per the [GraphQL Kotlin docs](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/customizing-schemas/restricting-input-output/), we can use `@GraphQLValidObjectLocations` to restrict the use of a schema class as an input or output. With schema-first development, we want to enforce that schema is defined first up front. Thus we always want to be intentional about how the generated classes can be used.

### :link: Related Issues
- #44

### CODE REVIEW

File: test/unit/should_generate_field_resolver_interfaces/expected.kt [hunk 1]
```diff
@@ -18,6 +18,7 @@ interface QueryCompletableFuture {
     fun nonNullableResolver(arg: InputTypeGenerateFieldResolverInterfaces): java.util.concurrent.CompletableFuture<String>
 }
 
+@GraphQLValidObjectLocations(locations = [GraphQLValidObjectLocations.Locations.INPUT_OBJECT])
```
Comment (ameerabdul) [inline test/unit/should_generate_field_resolver_interfaces/expected.kt:21]: why does this need this annotation INPUT_OBJECT. This doesn't end with `Input` and also this is an interface.

Comment (danadajian) [reply test/unit/should_generate_field_resolver_interfaces/expected.kt:21]: This class is generated from [this type](https://github.com/ExpediaGroup/graphql-kotlin-codegen/blob/main/test/unit/should_generate_field_resolver_interfaces/schema.graphql#L8)

Comment (danadajian) [reply test/unit/should_generate_field_resolver_interfaces/expected.kt:21]: Basically any `input` type that doesn't have a matching `type` is going to get this annotation

Comment (danadajian) [reply test/unit/should_generate_field_resolver_interfaces/expected.kt:21]: And vice versa - any `type` that doesn't have a matching `input` will get `@GraphQLValidObjectLocations(locations = [GraphQLValidObjectLocations.Locations.OBJECT])`

Comment (danadajian) [reply test/unit/should_generate_field_resolver_interfaces/expected.kt:21]: Will add detailed docs on this later

**Review by ameerabdul — APPROVED** [review]

**Pull request merged**